src/App.tsx

```diff
@@ -1,7 +1,30 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    lastKey: '',
+    listener: (event: KeyboardEvent) => {
+      this.setState({ lastKey: event.key });
+    },
+  };
+
```

Comment:
not a good idea to put the callback inside the state - it's not the state by nature
```suggestion
  state = {
    lastKey: '',
  };
  
  listener = (event: KeyboardEvent) => {
    this.setState({ lastKey: event.key });
  }
  // i think you'll just need to declare it below lifecycle methods
  // and then you just use it like `this.listener` on adding and removing listeners

```